homeassistant/components/rova/sensor.py

```diff
@@ -27,36 +31,47 @@
 SCAN_INTERVAL = timedelta(hours=12)
 
 
-class RovaSensorMetadata(NamedTuple):
+@dataclass
+class RovaSensorEntityDescription(SensorEntityDescription):
     """Metadata for an individual rova sensor."""
 
-    name: str
-    json_key: str
-    icon: str
+    json_key: str | None = None
 
+    def __post_init__(self) -> None:
+        """Check required keys are present."""
+        if self.json_key is None:
+            # pragma: no cover
+            raise TypeError
+        self._json_key = self.json_key
```

Comment:
My current solution for the required key problem. `self._json_key` will be inferred as just `str`.